### PR TITLE
Add compile option to remove symbol tables, debugging info, and path info from the Go binary

### PR DESCRIPTION
**Pull request by cw75**
The `-lean` Make option does the trick. We should use this flag to compile the binaries and upload them to S3.

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -21,6 +21,21 @@ server:
 	mkdir -p build
 	cp $$GOPATH/bin/server build/server
 
+migrator-lean:
```
Comment (vsreekanti) [inline src/Makefile:24]: Tiny, tiny naming nit -- the `lean` flag is an implementation detail. Can we call this `make migrator-release` and then add a target called `release` which is just a dummy target for all 3 of these?

**Review by vsreekanti — APPROVED** [review]

**Comment by vsreekanti** [comment]
@cw75, out of curiosity, I don't see `-lean` anywhere in the codebase. How is this specified?

**Comment by cw75** [comment]
@vsreekanti Oh I was planning to run `make release` manually to compile and then upload to S3. So there's no where else in the repo that uses this flag. Do you have anything else in mind?

**Comment by vsreekanti** [comment]
Nope, just curious. This LGTM!

**Pull request merged**